megamek/src/megamek/utils/RATGeneratorEditor.java

```diff
@@ -55,24 +56,28 @@
 
     private final JTabbedPane panMain = new JTabbedPane();
 
-    private final JTextField txtSearch = new JTextField();
+    private final JTextField txtSearch = new JTextField(20);
     private final JTable tblMasterUnitList = new JTable();
     private MasterUnitListTableModel masterUnitListModel;
     private TableRowSorter<MasterUnitListTableModel> masterUnitListSorter;
 
-    private final JTextField txtFaction = new JTextField();
+    private final JComboBox<FactionRecord> factionChooser = new JComboBox<>();
```

Comment:
For future code, I'd recommending using MMComboBox as that'll simplify the implementation.